File: estrutura_de_dados/ordenacao_basica/C++/insertion_sort.cpp
#include <iostream>

using namespace std;

void insertion_sort(int vetor[], int tam){
    for(int i = 1; i < tam; i++){
        int menor_valor = vetor[i];
        int j = (i - 1);

        while((j >= 0) && (vetor[j] > menor_valor)){
            vetor[j + 1] = vetor[j];
            j = (j - 1);
        }

        vetor[j + 1] = menor_valor;
    }
}

int main(){
    int vetor[] = {561, 165, 10, 88, 0, 1, 2, 55, 545, 0, -1};
    int tam = 11;

    for(int i = 0; i < tam; i++){
        cout << vetor[i] << " ";
    }
    cout << " " << endl;

    insertion_sort(vetor, tam);

    for(int i = 0; i < tam; i++){
        cout << vetor[i] << " ";
    }
}

File: estrutura_de_dados/Java/src/metodos_de_busca/InsertionSort.java
package metodos_de_busca;

public class InsertionSort {
	static void insertion_sort(int vetor[]) {
		for(int i = 1; i < vetor.length; i++) {
			int menor_valor = vetor[i];
			int j = (i - 1);
			
			while((j >= 0) && (vetor[j] > menor_valor)) {
				vetor[j + 1] = vetor[j];
				j = (j - 1);
			}
			
			vetor[j + 1] = menor_valor;
		}
	}
	
	public static void main(String[] args) {
		int vetor[] = {545, 545, 51, 155, 215, 5161, 511665, 0, 1, 2, 3, 5, 8, 55, 88};
		
		for(int i = 0; i < vetor.length; i++) {
			System.out.print(vetor[i] + " ");
		}
		System.out.println("");
		
		insertion_sort(vetor);
		
		for(int i = 0; i < vetor.length; i++) {
			System.out.print(vetor[i] + " ");
		}
	}
}
